### PR TITLE
Add a function to normalize density matrix

### PR DESCRIPTION
**Pull request by mscho527**
_Related Issue: #885_

This small PR adds a small function to `density` that normalizes the density matrix using `numpy.linalg.norm`. This is necessary for DDEC6 implementation.

### CODE REVIEW

File: cclib/method/density.py [hunk 1]
```diff
@@ -73,15 +73,37 @@ def calculate(self, fupdate=0.05):
...
+        if not unrestricted:  # multiply by two to account for second electron
             self.density[0] = numpy.add(self.density[0], self.density[0])
 
         if self.progress:
             self.progress.update(nstep, "Done")
 
-        return True #let caller know we finished density
+        return True  # let caller know we finished density
+
+    def normalize(self):
```
Comment (langner) [inline cclib/method/density.py:88]: Unit test?

File: cclib/method/density.py [hunk 2]
```diff
@@ -73,15 +73,37 @@ def calculate(self, fupdate=0.05):
...
+            return False
+
+        self.density[0] = self.density[0] / norm
+
+        if len(self.density) == 2:
+            # Normalize beta density matrix as well
+            norm = numpy.linalg.norm(self.density[1])
+
+            if norm == 1:
+                self.logger.error("Norm of density matrix equals zero.")
```
Comment (langner) [inline cclib/method/density.py:104]: Make comment specific to beta.

File: cclib/method/density.py [hunk 3]
```diff
@@ -73,15 +73,37 @@ def calculate(self, fupdate=0.05):
...
+            self.logger.error("Norm of density matrix equals zero.")
+            return False
+
+        self.density[0] = self.density[0] / norm
+
+        if len(self.density) == 2:
+            # Normalize beta density matrix as well
+            norm = numpy.linalg.norm(self.density[1])
+
+            if norm == 1:
```
Comment (langner) [inline cclib/method/density.py:103]: == 0

File: cclib/method/density.py [hunk 4]
```diff
@@ -73,15 +73,37 @@ def calculate(self, fupdate=0.05):
...
+                self.logger.error("Norm of density matrix equals zero.")
+                return False
+
+                self.density[1] = self.density[1] / norm
```
Comment (langner) [inline cclib/method/density.py:107]: Unindent.

**Comment by mscho527** [comment]
Closing this PR since I've realized that this wouldn't be necessary for DDEC6 calculation -- only normalization checks are needed for one-particle density matrices. This should just be a check within the implementation with an `assert` type statement.

**Pull request closed without merging**